### PR TITLE
testing of step-40 to step-49

### PR DESCRIPTION
**Pull request by vyushut**
Proceeding after #17100, #17084, #17130.

CTests should not write any vtk and, if they output any iteration counts, they should do it with with a range of acceptable answers.

### CODE REVIEW

File: tests/examples/step-42.diff [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+<       pcout << "         Error: " << solver_control.initial_value() << " -> "
+<             << solver_control.last_value() << " in "
+<             << solver_control.last_step() << " Bicgstab iterations."
+<             << std::endl;
+---
+> //      solver.solve(newton_matrix,
+> //                   distributed_solution,
+> //                   newton_rhs,
+> //                   preconditioner);
+>       check_solver_within_range(std::cout,
```
Comment (tjhei) [inline tests/examples/step-42.diff:20]: this needs to be 
```suggestion
>       check_solver_within_range(pcout,
```

File: tests/examples/step-44.diff [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+> //        std::cout << " | " << std::fixed << std::setprecision(3) << std::setw(7)
+> //                  << std::scientific << lin_solver_output.first << "  "
+> //                  << lin_solver_output.second << "  "
+> //                  << error_residual_norm.norm << "  " << error_residual_norm.u
+> //                  << "  " << error_residual_norm.p << "  "
+> //                  << error_residual_norm.J << "  " << error_update_norm.norm
+> //                  << "  " << error_update_norm.u << "  " << error_update_norm.p
+> //                  << "  " << error_update_norm.J << "  " << std::endl;
+1946c1946
+<     std::cout << " ASM_SYS " << std::flush;
```
Comment (tjhei) [inline tests/examples/step-44.diff:68]: I think these outputs you can leave as is

File: tests/examples/step-48.diff [hunk 3]
```diff
@@ -1,16 +1,22 @@
-59c59
+58c58
 <   // it is passed as a template argument to the implementation of the
```
Comment (tjhei) [inline tests/examples/step-48.diff:2]: why this change?

File: tests/examples/step-48.diff [hunk 4]
```diff
@@ -24,28 +30,30 @@
...
+618,623d596
 < 
 <     pcout << "   Average wallclock time per time step: "
 <           << wtime / timestep_number << 's' << std::endl;
 < 
 <     pcout << "   Spent " << output_time << "s on output and " << wtime
 <           << "s on computations." << std::endl;
-645c618
+644c617
 <     argc, argv, numbers::invalid_unsigned_int);
```
Comment (tjhei) [inline tests/examples/step-48.diff:57]: why this change?

File: tests/examples/step-49.diff [hunk 5]
```diff
@@ -0,0 +1,15 @@
+88,91c88,92
+<   std::ofstream out(filename);
+<   GridOut       grid_out;
+<   grid_out.write_vtu(triangulation, out);
```
Comment (tjhei) [inline tests/examples/step-49.diff:4]: only remove this single line

File: tests/examples/step-40.diff [hunk 6]
```diff
@@ -17,11 +17,15 @@
...
 >                               8,
 >                               15);
-634c636
+643c645
 <     const unsigned int n_cycles = 8;
 ---
 >     const unsigned int n_cycles = 3;
-662c664
+659c661
+<         output_results(cycle);
```
Comment (tjhei) [inline tests/examples/step-40.diff:25]: I thought about this some more and I think we should keep all calls to graphical output.

File: tests/examples/step-43.diff [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+>                                 solver_control.last_step(),
+>                                 15, 19);
+1594,1595c1603,1604
+<       std::cout << "        ..." << solver_control.last_step()
+<                 << " CG iterations." << std::endl;
+---
+> //      std::cout << "        ..." << solver_control.last_step()
+> //                << " CG iterations." << std::endl;
+2161,2162c2170,2171
+<         if (timestep_number % 200 == 0)
```
Comment (tjhei) [inline tests/examples/step-43.diff:45]: keep these

File: tests/examples/step-44.diff [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+---
+> //    output_results();
+1114c1114
+<     while (time.current() < time.end())
+---
+>     while (time.current() < 0.3)
+1126c1126
+<         output_results();
+---
+> //        output_results();
```
Comment (tjhei) [inline tests/examples/step-44.diff:20]: and these

File: tests/examples/step-44.diff [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+1543c1543
+<     std::cout << "    Setting up quadrature point data..." << std::endl;
```
Comment (tjhei) [inline tests/examples/step-44.diff:22]: why do you remove this?

File: tests/examples/step-45.diff [hunk 10]
```diff
@@ -0,0 +1,4 @@
+768c768
+<         output_results(refinement_cycle);
```
Comment (tjhei) [inline tests/examples/step-45.diff:2]: keep this

File: tests/examples/step-46.diff [hunk 11]
```diff
@@ -0,0 +1,6 @@
+1011,1012c1011,1012
+<         std::cout << "   Writing output..." << std::endl;
+<         output_results(refinement_cycle);
+---
+>          std::cout << "   Writing output..." << std::endl;
```
Comment (tjhei) [inline tests/examples/step-46.diff:5]: keep these

File: tests/examples/step-47.diff [hunk 12]
```diff
@@ -0,0 +1,4 @@
+905c905
+<         output_results(cycle);
```
Comment (tjhei) [inline tests/examples/step-47.diff:2]: keep these

File: tests/examples/step-49.diff [hunk 13]
```diff
@@ -0,0 +1,9 @@
+91,91c91,92
+<   std::cout << " written to " << filename << std::endl << std::endl;
+---
+> //  std::cout << " written to " << filename << std::endl << std::endl;
```
Comment (tjhei) [inline tests/examples/step-49.diff:4]: keep this

File: tests/examples/step-45.diff (hunk 10 above)
Comment (tjhei) [reply tests/examples/step-45.diff:2]: Keep this line

**Review by tjhei — APPROVED** [review]

**Review by kronbichler — APPROVED** [review]
